sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -604,7 +603,6 @@ private[hive] class HiveClientImpl(
     // these properties are still available to the others that share the same Hive metastore.
     // If users explicitly alter these Hive-specific properties through ALTER TABLE DDL, we respect
     // these user-specified values.
-    verifyColumnDataType(table.dataSchema)
```

Comment:
Any side-effect from this removal in `alterTable`?

Reply:
This removal prevents Spark from rechecking incompatible hive types from table properties. However, those errors before or after this change, are captured in HiveExternalCatalog to fallback, so the behavior keeps the same.

BTW, I do have a follow-up to optimize the caller side(alterTable) from HiveExternalCatalog to do an explicit 'incompatible hive types' check like what has been done in createTable.